examples/examples/tower-load.rs

```diff
@@ -1,410 +1,379 @@
+                            rsp(StatusCode::INTERNAL_SERVER_ERROR, error)
+                        }
+                        Ok(()) => rsp(StatusCode::NO_CONTENT, Body::empty()),
+                    }
+                }
+                _ => rsp(StatusCode::NOT_FOUND, "try `/filter`"),
+            };
+            Ok(rsp)
+        };
+        Box::pin(f)
+    }
+}
+
+impl<S> AdminSvc<S>
+where
+    S: tracing::Subscriber + 'static,
+{
+    fn set_from(&self, bytes: Bytes) -> Result<(), String> {
+        use std::str;
+        let body = str::from_utf8(&bytes.as_ref()).map_err(|e| format!("{}", e))?;
+        trace!(request.body = ?body);
+        let new_filter = body
+            .parse::<tracing_subscriber::filter::EnvFilter>()
+            .map_err(|e| format!("{}", e))?;
+        self.handle.reload(new_filter).map_err(|e| format!("{}", e))
+    }
+}
+
+fn rsp(status: StatusCode, body: impl Into<Body>) -> Response<Body> {
+    Response::builder()
+        .status(status)
+        .body(body.into())
+        .expect("builder with known status code must not fail")
+}
+
+impl HandleError {
+    fn bad_request(e: impl std::error::Error + Send + 'static) -> Self {
+        HandleError::BadRequest(Box::new(e))
+    }
+}
+
+impl fmt::Display for HandleError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            HandleError::BadPath => f.pad("path must be a single ASCII character"),
+            HandleError::NoContentLength => f.pad("request must have Content-Length header"),
+            HandleError::BadRequest(ref e) => write!(f, "bad request: {}", e),
+            HandleError::Unknown => f.pad("unknown internal error"),
+        }
+    }
+}
+
+impl std::error::Error for HandleError {}
+
+impl fmt::Display for WrongMethod {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "unsupported method: please use one of {:?}", self.0)
+    }
+}
+
+impl std::error::Error for WrongMethod {}
+
+fn gen_uri(authority: &str) -> (usize, String) {
+    static ALPHABET: &str = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    let mut rng = rand::thread_rng();
+    let idx = rng.gen_range(0, ALPHABET.len() + 1);
+    let len = rng.gen_range(0, 26);
+    let letter = ALPHABET.get(idx..=idx).unwrap_or("");
+    (len, format!("http://{}/{}", authority, letter))
+}
+
+// .instrument(tracing::info_span!(target: "gen", "load_gen", remote.addr=%addr));
+#[tracing::instrument]
+async fn load_gen(addr: &SocketAddr) -> Result<(), Err> {
+    let svc = ServiceBuilder::new()
+        .buffer(5)
+        .layer(request_span::layer(req_span))
+        .timeout(Duration::from_millis(200))
+        .service(Client::new());
+
+    while let Some(_) = time::interval(Duration::from_millis(50)).next().await {
+        let authority = format!("{}", addr);
+        let mut svc = svc.clone();
+        svc.ready().await?;
+        let f = async move {
```

Comment:
the original version of the example had a span for each _request_ with the request's message and path:
https://github.com/tokio-rs/tracing/blob/65547d8809fcc726b8187db85b23c42e32ef5dce/examples/examples/tower-load.rs#L372-L377

the idea was that you would use this span to figure out which paths were receiving which error codes. now, you can't really do that anymore. can we bring this back?

Reply:
also, we used to have a "sending request" event here. the idea was this would show how stuff might occur in between when you logged a message that you were sending a request and when you got back the response, and there might be interleaved messages, but that spans could be used to associate the requests and responses. now we can't really demo this any longer, because we only see the "response complete" events in the load generator...

Reply:
Done, I think.

Reply:
i think we need to avoid the issues with `span.enter`, but otherwise, yeah.